src/stream.h

```diff
@@ -3,6 +3,7 @@
  * @brief todo
  */
 #pragma once
+#include <utility>
```

Comment:
There are a lot of includes for this. Are they really needed? Especially here, in the header.

Reply:
Yes. Compilation fails otherwise.

Reply:
It seems they are needed before any boost includes, otherwise a lot of errors like this are observed:
https://github.com/tez011/Sunshine/actions/runs/8484576885/job/23247793427

If we can include a boost header in here, we can certainly include a system header above it, no?

Reply:
No problems with that, it's just that currently include state in Sunshine is a mess. For example, in this file even include `asio`, but it not used anywhere...

But that's out of scope for this PR.